tests/static-analysis/Type/intersection.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+use Psl\Collection\Map;
+use Psl\Result\ResultInterface;
+use Psl\Type;
+
+/**
+ * @psalm-suppress UnusedParam
+ *
+ * @param Map&ResultInterface&stdClass $value
+ */
+function takes_valid_intersection($value): void
+{
+}
+
+function test(): void
+{
+    /** @psalm-suppress MissingThrowsDocblock */
+    $old_school_codec = Type\intersection(
+        Type\object(Map::class),
+        Type\intersection(
+            Type\object(ResultInterface::class),
+            Type\object(stdClass::class),
+        ),
+    );
+
+    /** @psalm-suppress MissingThrowsDocblock */
+    $new_codec = Type\intersection(
+        Type\object(Map::class),
+        Type\object(ResultInterface::class),
+        Type\object(stdClass::class),
+    );
```

Comment:
hm, can you test this with 4 objects? i don't think it's going to work....

Reply:
hm, it seems to work here https://psalm.dev/r/433e9e351b, not sure why i thought it wouldn't ...

Reply:
Sure, also check https://psalm.dev/r/e1d0238b41